### PR TITLE
fix: test and fix untested method createNewClass()

### PR DESCRIPTION
**Pull request by monperrus**

### CODE REVIEW

File: src/main/java/spoon/reflect/factory/CodeFactory.java [hunk 1]
```diff
@@ -202,21 +204,21 @@ public CodeFactory(Factory factory) {
...
-	 *
-	 * @param type the decelerating type of the constructor.
-	 * @param anonymousClass Anonymous class in the new class.
-	 * @param parameters the arguments of the constructor call.
-	 * @param <T> the actual type of the decelerating type of the constructor if available/
-	 * @return the new class.
+	 * see {@link #createNewClass(CtClass, CtExpression[])}
 	 */
-	public <T> CtNewClass<T> createNewClass(CtTypeReference<T> type, CtClass<?> anonymousClass, CtExpression<?>...parameters) {
+	public <T> CtNewClass<T> createNewClass(CtType<T> superClass, CtExpression<?>...parameters) {
```
Comment (surli) [inline src/main/java/spoon/reflect/factory/CodeFactory.java]: I'm surprised that revapi is not complaining there?

**Comment by monperrus** [comment]
gakoci/revapi is dead since mid january :-)

**Comment by monperrus** [comment]
ready to merge

File: src/main/java/spoon/reflect/factory/CodeFactory.java (hunk 1 above)
Comment (surli) [reply src/main/java/spoon/reflect/factory/CodeFactory.java]: I'm not a huge fan on this signature change: it's really a potential breaking change for users. Why don't you keep the existing one and add a new one?

Comment (monperrus) [reply src/main/java/spoon/reflect/factory/CodeFactory.java]: This method 1) was useless and incorrect 1) is not used anywhere (consequence of 1))

So I'd say there is no need to keep this technical debt.

**Pull request merged**